### ISSUE TITLE
Shared documents: Clicking on heading anchors should copy a link to the clipboard

### ISSUE THREAD

**Issue by tommoor** [outline/outline#815]
Pop up a little toast message after

**Comment by satyadeepk** [comment]
Looks like it's already happening.

<img src="https://user-images.githubusercontent.com/478578/49683507-37c9d480-faec-11e8-9108-c9405ad49562.png" width=400/>

**Comment by tommoor** [comment]
Ah, thanks for the note – looks like it isn't working on publicly shared documents only 😄

**Comment by satyadeepk** [comment]
I see. I have some time this weekend. Will see if it's low hanging.

**Comment by satyadeepk** [comment]
I noticed that the link is actually being copied in share page, but the toast is not visible. 
It's because, the `Toast` component is defined in `Layout`, but the `Layout` is not included in the share `Route`. `Layout` is only available in `Authenticated` section.

```
export default function Routes() {
  return (
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/share/:shareId" component={Document} />
      <Authenticated>
        <Layout>
          <Switch>
            <Route path="/dashboard/:tab" component={Dashboard} />
            <Route path="/dashboard" component={Dashboard} />
```

Also, another issue I noticed is that the copied URLs are actually not scrolling to the anchors when opening in new tab. This happens both for `/doc/` and `/share/` links. 
However, they are able to scroll when the URL is pasted in the same tab.

Looks like the hash is eaten up before the control reaches `ScrollToAnchor`.
